### PR TITLE
Migrate v1 page

### PR DESCRIPTION
**Pull request by moodysalem**
Moves the migrate v1 liquidity feature to the frontend, reachable from two different places:
1. migrate liquidity link in the top bar
2. if you have liquidity in any v1 pool for a token in the default list or an added token, a 'migrate your v1 liquidity' link replaces the 'import pool' link

After this is merged, we should:
1. redirect migrate.uniswap.exchange to uniswap.exchange/migrate/v1
2. delete migrate branch

### CODE REVIEW

File: src/pages/MigrateV1/MigrateV1Exchange.tsx [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+  const v2Pair = usePair(WETH[chainId as ChainId], token)
+
+  const v2SpotPrice = v2Pair?.reserveOf(token)?.divide(v2Pair?.reserveOf(WETH[chainId as ChainId]))
+
+  const [confirmingMigration, setConfirmingMigration] = useState<boolean>(false)
+  const [pendingMigrationHash, setPendingMigrationHash] = useState<string | null>(null)
+
+  const sharePercent =
+    exchangeETHBalance && totalSupply ? new Percent(liquidityTokenAmount.raw, totalSupply.raw) : ZERO_FRACTION
```
Comment (NoahZinsmeister) [inline src/pages/MigrateV1/MigrateV1Exchange.tsx:47]: why isn't this just `liquidityTokenAmount && totalSupply`?

File: src/pages/MigrateV1/MigrateV1Exchange.tsx [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+
+  const ethWorth: Fraction = exchangeETHBalance
+    ? new Fraction(sharePercent.multiply(exchangeETHBalance).quotient, WEI_DENOM)
+    : ZERO_FRACTION
+
+  const tokenWorth: TokenAmount = exchangeTokenBalance
```
[the remark this answers is not in the document]
Comment (moodysalem) [reply src/pages/MigrateV1/MigrateV1Exchange.tsx:52]: `shareFraction` (now) always defined

File: src/pages/MigrateV1/MigrateV1Exchange.tsx [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+  const v2SpotPrice = v2Pair?.reserveOf(token)?.divide(v2Pair?.reserveOf(WETH[chainId as ChainId]))
+
+  const [confirmingMigration, setConfirmingMigration] = useState<boolean>(false)
+  const [pendingMigrationHash, setPendingMigrationHash] = useState<string | null>(null)
+
+  const sharePercent =
+    exchangeETHBalance && totalSupply ? new Percent(liquidityTokenAmount.raw, totalSupply.raw) : ZERO_FRACTION
+
+  const ethWorth: Fraction = exchangeETHBalance
```
[the remark this answers is not in the document]
Comment (moodysalem) [reply src/pages/MigrateV1/MigrateV1Exchange.tsx:48]: `shareFraction` (now) always defined

File: src/pages/MigrateV1/MigrateV1Exchange.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (moodysalem) [reply src/pages/MigrateV1/MigrateV1Exchange.tsx]: `liquidityTokenAmount` is a required prop, removed the `exchangeETHBalance` from ternary

**Pull request merged**